### DISCUSSION TITLE
-x --x-arg for commands other than upgrade/downgrade/migrate

### DISCUSSION THREAD

**Discussion by xmikew** [sqlalchemy/alembic#1313] [Usage Questions] [answered]
When using the --multidb or where you have many alembic_version tables across multiple dbs managed by one alembic environment, it would be nice to be able to run stamp/check etc on selected databases. I enable this using the -x --x-args and then in env.py filtering the engines based on the input for the commands that do accept -x --x-arg

Does it make sense for stamp/current etc to also accept  -x --x-arg or am I missing something? Presently, stamp/current runs on all dbs which was surprising when I ran stamp and set all the databases to head

**Comment by zzzeek** [comment]
all commands accept the -x argument, it's part of the global options on the argparse object.    there's also the issue of which commands make use of `env.py`, but any commands that access the database will be using `env.py` by definition, like stamp, current, etc.

It sounds like you have something that's not actually working as expected, so this should be a discussion first.

**Reply by xmikew** [reply]
Thanks zzzeek,

I have multi-tenant app using Flask-migrate

I think things are working as expected? Sometimes in testing one (or more) of my database drifts so far from models, that I just want to drop_all, create_all on it again and then just stamp that database at head.

Also sometimes I will test an upgrade/roll-out migrations to a subset of databases first. I pass the ones I want to do via -x -x-arg and in my env.py skip the online migration unless selected. So running check with subset of databases would be nice so I know which ones of the subset I've done.

I did not see -x available when running alembic check --help or alembic stamp --help. passing in -x to either of these causes an error and the command does not run. alembic 1.11.1. I'm running 'flask db check -x db --x-arg dbname1,dbname2

Thanks!

**Reply by CaselIT** [reply, marked as answer]
`-x` is an argument of the alembic base command, so it will be shown only if you do `alembic --help`.

>  passing in -x to either of these causes an error and the command does not run. alembic 1.11.1. I'm running 'flask db check -x db --x-arg dbname1,dbname2

passing `-x` like that will not work also when using upgrade or downgrade: running `alembic upgrade -x foo head` will also error with `alembic: error: unrecognized arguments: -x head`
Like in the other cases you will have to place -x before specifying the command to pass alembic

**Reply by xmikew** [reply]
Thanks, looks like it's the integration between Flask-Migrate to alembic then. I'll check there instead. Thanks for your patience and direction!
